test/distributed/test_c10d_nccl.py

```diff
@@ -3529,6 +3529,61 @@ def test_new_group_local_sync_sanity_check(self):
     def test_new_group_local_sync_duplicated_pg(self):
         self._test_new_group_local_sync_duplicate_pg(backend="nccl")
 
+    @requires_nccl()
+    @skip_if_lt_x_gpu(4)
+    def test_gather_subgroup(self):
+        if self.rank > 3:
```

Comment:
do we want to set world_size explicitly? right now it's hard coded to 4 from from ``test_c10d_common.AbstractLargeCommTest`` any way

```
@property
    def world_size(self):
        return 4
```

Reply:
i wrote the test in as simple a way as possible, hardcoding lists of ranks e.g. [0, 1] instead of doing things like `ranks = list(range(self.world_size))[:self.world_size/2]`.  It just felt easier to read and even less LOC this way.  It is not that important to have the test run on 8 gpus if they are available.

Reply:
thanks. it's clear now with ``world_size=4`` in ``init_process_group``